xarray_sentinel/sentinel1.py

```diff
@@ -141,34 +142,115 @@ def open_orbit_dataset(product_path: PathType) -> xr.Dataset:
-    _, manifest = esa_safe.open_manifest(product_path)
-    product_attrs, product_files = esa_safe.parse_manifest_sentinel1(manifest)
-    sub_swaths = product_attrs["xs:instrument_mode_swaths"]
-    groups_lev0 = sub_swaths
-    groups_lev1 = []
-    for sub_swath in sub_swaths:
-        for data in METADATA_OPENERS:
-            groups_lev1.append(f"{sub_swath}/{data}")
-    return groups_lev0, groups_lev1
-
-
-def open_root_dataset(product_path: str) -> xr.Dataset:
-    _, manifest = esa_safe.open_manifest(product_path)
-    product_attrs, product_files = esa_safe.parse_manifest_sentinel1(manifest)
-    sub_swaths = product_attrs["xs:instrument_mode_swaths"]
-    product_attrs["groups"] = []
-    for sub_swath in sub_swaths:
-        for data in METADATA_OPENERS:
-            product_attrs["groups"].append(f"{sub_swath}/{data}")
+def find_avalable_groups(
+    manifest_path: PathType,
+    product_attrs: T.Dict[str, T.Any],
+    product_files: T.Dict[str, str],
+) -> T.Dict[str, T.Dict[str, T.Any]]:
+    ancillary_data_paths = esa_safe.get_ancillary_data_paths(
+        manifest_path, product_files
+    )
+    groups: T.Dict[str, T.Dict[str, T.Any]] = {}
+    for subswath_id, subswath_data in ancillary_data_paths.items():
+        subswath_id = subswath_id.upper()
+        burst_info = get_burst_info(product_attrs, subswath_id, subswath_data)
+        subgroups = list(METADATA_OPENERS.keys()) + list(burst_info.keys())
+        groups[subswath_id] = dict(subswath_data, subgroups=subgroups)
+        for subgroup in METADATA_OPENERS.keys():
+            groups[f"{subswath_id}/{subgroup}"] = subswath_data
+        for burst_id, burst_data in burst_info.items():
+            full_data = dict(subswath_data, **burst_data)
+            groups[f"{subswath_id}/{burst_id}"] = full_data
+    return groups
+
+
+def open_root_dataset(
+    product_attrs: T.Dict[str, str], groups: T.Dict[str, T.Dict[str, T.Collection[str]]]
+) -> xr.Dataset:
+    product_attrs = dict(product_attrs, groups=list(groups.keys()))
     return xr.Dataset(attrs=product_attrs)  # type: ignore
 
 
-def open_swath_dataset(product_path: str, group: str) -> xr.Dataset:
-    product_attrs = {"groups": ["orbit", "attitude", "gcp"]}
+def open_swath_dataset(
+    product_attrs: T.Dict[str, str],
+    swath_id: str,
+    swath_data: T.Dict[str, T.Collection[str]],
+) -> xr.Dataset:
+    product_attrs = dict(product_attrs, groups=swath_data["subgroups"])
     return xr.Dataset(attrs=product_attrs)  # type: ignore
 
 
+def open_burst_dataset(
+    product_attrs: T.Dict[str, str], burst_id: str, burst_data: T.Dict[str, T.Any]
+) -> xr.Dataset:
+    ds = xr.merge(
+        [
+            {
+                pol.upper(): rioxarray.open_rasterio(datafile)
+                for pol, datafile in burst_data["measurement"].items()
+            }
+        ]
+    )
+    ds.attrs.update(product_attrs)
+    ds = ds.squeeze("band").drop(["band", "spatial_ref"])
+    return ds.isel(
```

Comment:
Code style. We don't put expression longer than one line in `return`. `return` should be on the last line of a function.

Reply:
Fixed in 9455afbf461295597bebf417a99983c57ae9a0b2